src/settings/MappingProfiles/EditMappingProfileRoute/EditMappingProfileRoute.js

```diff
@@ -0,0 +1,92 @@
+import React, { useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import {
+  get,
+  omit,
+} from 'lodash';
+
+import { stripesConnect } from '@folio/stripes-core';
+import {
+  FullScreenView,
+  Preloader,
+} from '@folio/stripes-data-transfer-components';
+
+import { useProfileHandlerWithCallout } from '../../utils/useProfileHandlerWithCallout';
+import { MappingProfilesFormContainer } from '../MappingProfilesFormContainer';
+import { mappingProfileTransformations } from '../MappingProfilesTransformationsModal/TransformationsField/transformations';
+import { generateTransformationFieldsValues } from '../MappingProfilesTransformationsModal/TransformationsField';
+
+const contentLabelId = 'ui-data-export.mappingProfiles.editProfile';
+
+export const EditMappingProfileRouteComponent = ({
+  resources: { mappingProfile },
+  mutator,
+  history,
+  match: { params },
+  location: { search },
+}) => {
+  const intl = useIntl();
+  const contentLabel = intl.formatMessage({ id: 'ui-data-export.mappingProfiles.editProfile' });
+
+  const handleCancel = useCallback(
+    () => history.push(`/settings/data-export/mapping-profiles/view/${params.id}${search}`),
+    [history, params.id, search],
+  );
+
+  const handleSubmit = useProfileHandlerWithCallout({
+    errorMessageId: 'ui-data-export.mappingProfiles.editedErrorCallout',
+    successMessageId: 'ui-data-export.mappingProfiles.editedCallout',
+    isCanceledAfterError: true,
+    onAction: mutator.mappingProfile.PUT,
+    onCancel: handleCancel,
+  });
+
+  const mappingProfileRecord = get(mappingProfile, 'records.0');
+
+  if (!mappingProfileRecord) {
```

Comment:
Are we going to use the preloader on new mapping profile page after integration with a backend side?

Reply:
Yep. I think in this kind of form.

Reply:
So, in this case it should be placed in the form to prevent having this stuff in several places. @ViktorSoroka07 what do you think?

Reply:
As discussed we will reconsider this in the integration story.